beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -1023,7 +1266,10 @@ impl<E: EthSpec> PeerManager<E> {
     /// Prune peers in the following order:
     /// 1. Remove worst scoring peers
     /// 2. Remove peers that are not subscribed to a subnet (they have less value)
-    /// 3. Remove peers that we have many on any particular subnet
+    /// 3. Remove peers that we have many on any particular subnet, with some exceptions
+    ///    - Don't remove peers needed for data column sampling (≥ MIN_SAMPLING_COLUMN_SUBNET_PEERS)
+    ///    - Don't remove peers needed for sync committees (>=MIN_SYNC_COMMITTEE_PEERS)
+    ///    - Don't remove peers from the lowest density attestation subnets
     /// 4. Randomly remove peers if all the above are satisfied
```

Comment:
I think we are now violating this last point. 

Assuming no peers commit any offense (no downscoring), it looks like we have a minimal set of peers, i.e:
MIN_SAMPLING_COLUMN_SUBNET_PEERS * no_of_subnets.  Because if we fill up all the subnets, to this constant, then all of those peers are protected from pruning. 

If our --target-peers is less than this number, I think we no longer prune down to our target. 

The 4th point here, was to just select random peers to prune down to our target.